src/traccuracy/__init__.py

```diff
@@ -3,7 +3,7 @@
 
 try:
     __version__ = version("traccuracy")
-except PackageNotFoundError:
+except PackageNotFoundError:  # pragma: no cover
```

Comment:
A good question!  The tldr answer is "whenever you haven't run `pip install`".  For example, if you were developing locally, you hadn't run `pip install -e .`, you happened to `cd` into the src directory, and then ran `import traccuracy` :joy: ... it _would_ import in that case, but `importlib.metadata.version` would raise an Exception. read on for more...

----

This little bit of code here wraps a relatively complicated chain of tech that ultimately lets you tie your package versioning to your git tags (rather than a hard-coded string in your package).  Here's how it works (and I know some of this is review):
1. to mark a package, you tag a commit with an annotated tag `git tag -a 'vX.Y.Z'` 
2. when you run `python -m build` or `pip install .`, it reads the `[build-system]` section of your `pyproject.toml` and that installs  the [`setuptools_scm` package](https://github.com/pypa/setuptools_scm) in the build environment
3. setuptools-scm is a plugin that, at build time, will dynamically inspect the active git environment for the most recent tagged commit and inject it into the `project.version` field of your pyproject metadata.  (that's why you have `dynamic = ["version"]` in pyproject.toml).
4. (this is the bit that's often mysterious) that injected version string gets added into the metdata that goes along with your package.
    - you can see this locally: when you run `pip install -e .`, look at the file `src/traccuracy.egg-info/PKG-INFO` and you'll see a version based on tags:
        ```txt
        Metadata-Version: 2.1
        Name: traccuracy
        Version: 0.0.3.dev126+g706b164
        ```
    - when build for distribution to pypi, `python -m build` puts that tag into the `dist-info`.  Run `python-m build` locally, then unzip the `traccuracy...whl` file in `dist/` and you'll see a `dist-info` folder inside of it.  inside of that is a `METADATA` file with the same info as above
5.  When you pip install a package, that `dist-info` goes along with it and gets added to `site-packages`  
6. now... `importlib.metadata` is the standard library module for getting information from that `dist-info` folder for an *installed* package.  When you run `importlib.metadata.version('package-name')`, it looks for that dist-info (or editably installed `.egg-info`) folder and fetchs the version from the METADATA file.
7. if python can't find a `'package-name.dist-info'` folder anywhere, then it will raise an exception... it means the package simply hasn't been "installed".  
8. if it's unclear how you could even import a package that hasn't been installed, it's important to note that when you run `import thing`, all that matters is that there is a module or folder somewhere in `sys.path` by the name of `thing`.  It needn't be an "installed" package, it could simply be a `thing.py` file in the current working directory. 

make sense?  TMI? 😂

Reply:
and one more comment... if one wonders why go through all this nonsense for a version string...

It's not mandatory, you can definitely just hard-code a version in `__init__` and pull that dynamically at build time using many build tools,  including setuptools, which offers [`setuptools.dynamic.version.attr`](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata) to pull version from a file.  The main argument for using tags is that it's good practice.  it makes it much easier in the future to run `git checkout vX.Y.Z` (say, if you're trying to debug something that a user is having on a previous version).   Not that you couldn't _also_ git tag commits that had manual version strings in the `__init__` file.  but if you do, you're now duplicating the source of truth.  By using git tags as the single source of truth, you get the ability to `git checkout vX.Y.Z` and *know* that you're definitely getting the same thing as the `X.Y.Z` version that you uploaded to pypi, without the possibility of human error causing a mismatch

Reply:
I mostly figured out the dynamic version tagging stuff, although I now understand it in much more detail 🙂 and I like it! I was mostly curious about when the error would be thrown, but

>  you hadn't run pip install -e ., you happened to cd into the src directory, and then ran import traccuracy 😂 ...

 I guess explains it! Catching that potential error is a level of thoroughness I can only aspire to.

Re: versioning - should we be releasing more minor/patch versions on github as well to make this more useful? Is that supposed to happen whenever we merge a PR?

Thanks!!

Reply:
> Re: versioning - should we be releasing more minor/patch versions on github as well to make this more useful? Is that supposed to happen whenever we merge a PR?

people have all kinds of different opinions on the frequency of release, and the versioning scheme itself.  I hesitate to give any sort of prescriptive advice there. I personally like having packages in a state where you can basically release them anytime.  So, have lots of tests on your public API that ensure that all public functions are working, and then, within those bounds, feel free to bump your patch version as often as you need to fix things as they arise (it's great if you can just release a fix to someone's issue the next day... but that gets hard if releases start to feel like a major "deal" or something to be reckoned with)